examples/motors/motor_stepper.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef MOTOR_LINMOT_H
+#define MOTOR_LINMOT_H
+
+#include <time.h>
+#include <FastAccelStepper.h>
+
+#include "motor.hpp"
+
+FastAccelStepperEngine engine = FastAccelStepperEngine();
+
+class StepperMotor: public MotorInterface {
+  public:
+    // Motion
+    //void goToHome();
+    void goToPos(float position, float speed, float acceleration);
+    void stopMotion();
+
+    // General
+    void registerTasks();
+
+    // Stepper
+    void setStepsPerMm(int steps);
+    void setMovementPin(int stepPin, int dirPin);
+    void setEnablePin(int enablePin, bool activeHigh = true);
+
+    // TODO - Match this up with OSSM strategy for homing
+    // Assumes always homing to back of machine for safety - Is this valid?
+    void setSensorlessHoming(int currentPin, float currentLevel);
```

Comment:
Definitely. I'd always prefer sensor-less homing over some fieldbus protocol rather than an analog sensor. Especially when it's already integrated into the motor. Keeps part counts down, and the hardware simpler.

I do think there will be some people wanting Sensor-less homing via an Analog Current Sensor though. For example those with TB6600 drivers. I know those drivers were called out in OSSM, so I wouldn't be surprised if some people are using them instead of the JMC motors.

I'll move that to a generic motor_stepper.h file though. I might just leave it stubbed out and not implement it as you said too.

Reply:
In my experience everyone who started out with a stepper eventually spent the money on the servo. A perfect example of who buys cheap, buys twice. Who goes with the stepper should use a cheap limit switch. It's not like StrokeEngine would become unusable.